### PR TITLE
KAFKA-3663 : Implements KIP-59 for a kafka-brokers.sh command

### PR DESCRIPTION
**Pull request by JThakrar**
This implements KIP-59: Proposal for a kafka broker command.
See https://cwiki.apache.org/confluence/display/KAFKA/KIP-59%3A+Proposal+for+a+kafka+broker+command for details and sample output.

### CODE REVIEW

**Comment by JThakrar** [comment]
Hello @gwenshap ,

Can you please give this PR, the code and the Wiki documentation a look over?

Thanks,
Jayesh

> From: Gwen Shapira gwen@confluent.io
> To: dev@kafka.apache.org; Jayesh Thakrar j_thakrar@yahoo.com 
> Cc: Users users@kafka.apache.org
> Sent: Wednesday, May 11, 2016 12:26 PM
> Subject: Re: [DISCUSS] KIP-59 - Proposal for a kafka broker command - kafka-brokers.sh
> 
> Hello Jayesh,
> 
> Thank you for the suggestion. I like the proposal and the new tool seems useful.
> 
> Do you already have the tool available in a github repository?
> 
> If you don't, then this would be a good place to start - there are
> many Kafka utilities in github repositories (Yahoo's Kafka Manager as
> a famous example). This way users can evaluate the usefulness of the
> new tool before we include it in core-kafka.
> 
> Gwen

File: core/src/main/scala/kafka/admin/BrokerCommand.scala [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+import scala.collection._
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+import joptsimple._
+import org.apache.kafka.common.{Node, TopicPartitionInfo}
+import org.apache.kafka.clients.admin.{AdminClientConfig, AdminClient => JAdminClient}
+import org.apache.kafka.clients.admin.{DescribeClusterResult, DescribeTopicsResult, TopicDescription}
+import kafka.utils.{CommandLineUtils, Json}
+
+object BrokerCommand {
```
Comment (tedyu) [inline core/src/main/scala/kafka/admin/BrokerCommand.scala:31]: Can you add description for the tool along with sample usage ?

File: core/src/main/scala/kafka/admin/BrokerCommand.scala [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+    val nodes = cluster.nodes.get.asScala.toList
+    val topicList = adminClient.listTopics().names.get // All topics
+    val topicDescriptions = adminClient.describeTopics(topicList).values.asScala.values.map(_.get) // Partition details for all topics
+
+    val filteredNodes = filterNodes(nodes, brokers, hosts, racks)
+
+    val filteredTopicDescriptions: Iterable[TopicDescription] = filterTopicDescriptions(topicDescriptions, topics)
+
+    val mergedInfo = mergeNodeTopicDescriptions(filteredNodes, filteredTopicDescriptions)
+     //    nodeDetails.foreach(println)
```
Comment (tedyu) [inline core/src/main/scala/kafka/admin/BrokerCommand.scala:59]: Drop commented out code

**Review by tedyu — COMMENTED** [review]
Shouldn't org.apache.kafka.clients.admin.KafkaAdminClient be used ?

**Comment by ghost** [comment]
what's the status of this?

**Comment by darionyaphet** [comment]
Hi  @JThakrar  @tedyu  are you still working on this PR ?

**Comment by mimaison** [comment]
KIP-59 has never been voted and is abandoned so closing this PR. Feel free to restart the KIP discussion if it's still something you want to do.

**Pull request closed without merging**